Realm.Shared/Realm.cs

```diff
@@ -246,36 +246,50 @@ private void NotifyChanged(EventArgs e)
 
 
         /// <summary>
-        ///  Closes the Realm if not already closed. Safe to call repeatedly.
+        /// Closes the Realm if not already closed. Safe to call repeatedly.
+        /// Note that this will close the file. Other references to the same database
+        /// on the same thread will be invalidated.
         /// </summary>
-        [ReliabilityContract(Consistency.WillNotCorruptState, Cer.Success)]
         public void Close()
         {
             if (IsClosed)
                 return;
-            RuntimeHelpers.PrepareConstrainedRegions();
-            try { /* Close handle in a constrained execution region */ }
-            finally {
-                // Note we expect this call to also do explicit native close first rather than relying on destruction
-                // in case other handles preserve pointers - they will no longer work but don't stop closing
-                SharedRealmHandle.Close();
-            }
+
+            Dispose();
         }
 
+        ~Realm()
+        {
+            Dispose(false);
+        }
+
+        public void Dispose()
+        {
+            GC.SuppressFinalize(this);
+            Dispose(true);
+        }
 
         /// <summary>
         ///  Dispose automatically closes the Realm if not already closed.
         /// </summary>
-        public void Dispose()
+        public void Dispose(bool disposing)
```

Comment:
This should probably be private, as it will be confusing for end-users to have multiple dispose methods.